src/containers/profile/Profile.jsx

```diff
@@ -0,0 +1,152 @@
+import Header from "../common/Header";
+import "./Profile.css";
+import AccountSettings from "./ProfileSections/AccountSettings/AccountSettings";
+import ManageUsers from "./ProfileSections/ManageUsers/ManageUsers";
+import ManageBlogs from "./ProfileSections/ManageBlogs/ManageBlogs";
+import { updateProfilePhoto } from "../../actions/authActions";
+
+const Profile = (props) => {
+  const { profile, auth, id, currentUserType } = props;
+
+  const extraComponents = () => {
+    // Add available address details
+    let address = "";
+    if (profile.city && !profile.country) {
+      address = profile.city;
+    } else if (!profile.city && profile.country) {
+      address = profile.country;
+    } else {
+      address = `${profile.city}, ${profile.country}`;
+    }
+    const addressContainer = (
+      <div style={{ fontSize: "1.25em" }}>
+        <b>From </b>
+        <span>{address}</span>
+      </div>
+    );
+
+    return (
+      <div className="extraComponents">
+        <h1>
+          {profile.displayName ? profile.displayName : profile.fullName}
+          {profile.jobTitle && (
+            <span className="jobTitle">({profile.jobTitle})</span>
+          )}
+        </h1>
+        {(profile.city || profile.country) && addressContainer}
+        {profile.brief && <p className="profileBrief">{profile.brief}</p>}
+      </div>
+    );
+  };
+
+  // Protect the page from unauthorized access
+  if (!auth.uid) {
+    return <Redirect to="/login" />;
+  }
+
+  if (profile) {
+    if (!profile.isBlocked || currentUserType === "Admin") {
+      const profileImage = profile.imageURL ? profile.imageURL : anonymousImage;
+      return (
+        <div className="d-flex justify-content-center position-relative">
+          <Header
+            headerBackground={profileHeaderBackground}
+            profilePhoto={profileImage}
+            pageName={"profile"}
+            isOwner={id === auth.uid && true}
+            extraComponents={extraComponents()}
+            userId={auth.uid}
+            updateProfilePhoto={props.updateProfilePhoto}
+          />
+          <div className="shadow-lg pt-3 profileContainer">
+            <Tabs
+              defaultActiveKey="accountSettings"
+              className="justify-content-center"
+            >
+              {id === auth.uid && (
+                <Tab
+                  eventKey="accountSettings"
+                  title={<h4>Account Settings</h4>}
+                  className="m-4"
+                >
+                  <AccountSettings id={id} />
+                </Tab>
+              )}
+
+              {currentUserType === "Admin" && (
+                <Tab
+                  eventKey="manageUsers"
+                  title={<h4>Manage Users</h4>}
+                  className="m-4"
+                >
+                  <ManageUsers />
+                </Tab>
+              )}
+              {(profile.userType === "Author" ||
```

Comment:
sorry for that, in the beginning, I used only profile.userType but in case if a user's blocked I added a condition for admins to get specific info from the blocked user account